hazelcast-jet-core/src/main/java/com/hazelcast/jet/impl/execution/TaskletExecutionService.java

```diff
@@ -274,6 +266,7 @@ public void run() {
                 tracker.executionTracker.exception(new JetException("Exception in " + t + ": " + e, e));
             } finally {
                 blockingWorkerCount.decrementAndGet();
+                MetricsImpl.container().setContext(null);
```

Comment:
I would prefer to have a `clear()` method here, rather than setting it to null.

Reply:
Ok, I did that, but then it seems to me that I shouldn't any longer allow null `MetricsContext` param to be passed to `setContext`.  But if I do that I have to check for null, before calling `setContext` and that will make the caller code uglier. What do you think?